legion_test/bin/cleanup_nexus_repo

```diff
@@ -37,59 +37,56 @@ Options = NamedTuple('Options', [
 
 NEXUS_SEARCH_URL = '{host}/service/rest/v1/components'
 NEXUS_DELETE_URL = '{host}/service/rest/v1/components/{resource}'
-allowed_responce_codes = [200, 201, 202, 204]
+NEXUS_DEFAULT_RESOURCE_REPOSITORIES = 'jenkins_plugins', 'main-docker', 'pypi-hosted', 'helm-main'
+FIELDS_TO_ANALYZE = 'version', 'name'
+VERSION_SEARCH_REGEX = r'\d{12,14}'
+ALLOWED_RESPONSE_CODES = [200, 201, 202, 204]
 
 
-def get_builds(nexus_url, repository, user, password, build_name):
+def get_builds(repository: str, options: Options):
     """
     Get list of Legion artifacts at Nexus repo matching search filters
 
-    :param api_url: Nexus API URL
-    :type api_url: str
     :param repository: Nexus repository name
     :type repository: str
-    :param user: Nexus admin user
-    :type user: str
-    :param password: Nexus admin user password
-    :type password: str
-    :param build_name: Nexus component name, none for any
-    :return: list[dict, dict] List of Nexus components
+    :param options: configuration options
+    :type options: :py:class:`Options`
+    :return: list[dict] List of Nexus components
     """
-
-    search_filters = {"repository": repository}
-    if build_name:
-        search_filters["name"] = build_name
+    search_filters = {'repository': repository}
+    if options.build_name:
+        search_filters['name'] = options.build_name
 
     builds = []
     try:
-        url = NEXUS_SEARCH_URL.format(host=nexus_url)
+        url = NEXUS_SEARCH_URL.format(host=options.nexus_url)
         log.debug('Fetching Nexus artifacts from URL {}'.format(url))
-        response = requests.get(url, params=search_filters, auth=(user, password))
-        if response.status_code not in allowed_responce_codes:
-                raise Exception("Non 200 response from Nexus!")
+        response = requests.get(url, params=search_filters, auth=(options.user, options.password))
+        if response.status_code not in ALLOWED_RESPONSE_CODES:
+                raise Exception('Non 200 response from Nexus!')
```

Comment:
Please, add `resp.status code` and `resp.text` to the exception message.

Reply:
done